client/src/client_sync/v24/blockchain.rs

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Macros for implementing JSON-RPC methods on a client.
+//!
+//! Specifically this is methods found under the `== Blockchain ==` section of the
+//! API docs of Bitcoin Core `v24`.
+//!
+//! All macros require `Client` to be in scope.
+//!
+//! See or use the `define_jsonrpc_minreq_client!` macro to define a `Client`.
+
+/// Implements Bitcoin Core JSON-RPC API method `gettxspendingprevout`
+#[macro_export]
+macro_rules! impl_client_v24__get_tx_spending_prevout {
+    () => {
+        impl Client {
+            pub fn get_tx_spending_prevout(
+                &self,
+                outputs: &[bitcoin::OutPoint],
+            ) -> Result<GetTxSpendingPrevout> {
+                let json_outputs: Vec<_> = outputs.iter().map(|out| {
+                    serde_json::json!({
+                        "txid": out.txid.to_string(),
+                        "vout": out.vout,
+                    })
```

Comment:
Something is odd here, we should be able to use `into_json(out)`.

Reply:
Yeah, that's what I tried first but got the error "JSON value of type string is not of expected type object".

into_json(outputs) gives: 
```
Array [String("943682170364f0a4ea99f3364b4f2aa682343baae4c8b1eea15de788cb073bc8:0"), String("5069674b175134818f301fd51d41f55230c6ab592170f3d79e67b51610b328a7:0")]
```

`json_outputs` that works as an argument for the RPC is:
```
Array [Object {"txid": String("56b067fed4743116139e3c9a2e2cf2b6bf5c864a59fcfce4c8e92e896c1b2947"), "vout": Number(0)}, Object {"txid": String("3fcd727c4efbc0808597722bff13aed989eb8af2e1808ba62d09e05aa3581b9a"), "vout": Number(0)}]
```

Using into_json on the OutPoint directly make a single field `txid:vout` not separate like `json_outputs` above.